Framework/Algorithms/src/ConvertSpectrumAxis2.cpp

```diff
@@ -222,20 +247,24 @@ MatrixWorkspace_sptr ConvertSpectrumAxis2::createOutputWorkspace(
     newAxis->unit() = UnitFactory::Instance().create("QSquared");
   }
 
-  std::multimap<double, size_t>::const_iterator it;
   size_t currentIndex = 0;
-  for (it = m_indexMap.begin(); it != m_indexMap.end(); ++it) {
-    // Set the axis value.
-    newAxis->setValue(currentIndex, it->first);
-    // Copy over the data.
-    outputWorkspace->setHistogram(currentIndex, inputWS->histogram(it->second));
-    // We can keep the spectrum numbers etc.
-    outputWorkspace->getSpectrum(currentIndex)
-        .copyInfoFrom(inputWS->getSpectrum(it->second));
-    ++currentIndex;
-
-    progress.report("Creating output workspace...");
+
+  // Note that this is needed only for ordered case
+  if (m_toOrder) {
+    std::multimap<double, size_t>::const_iterator it;
+    for (it = m_indexMap.begin(); it != m_indexMap.end(); ++it) {
```

Comment:
Probably a minor point seeing as you have found a way to avoid this, but I think this loop could be multithreaded safely.